sklearn/decomposition/nmf.py

```diff
@@ -109,8 +109,8 @@ def _beta_divergence(X, W, H, beta, square_root=False):
         WH_data = _special_sparse_dot(W, H, X).data
         X_data = X.data
     else:
-        WH = np.dot(W, H)
-        WH_data = WH.ravel()
+        HW = np.dot(W, H)
```

Comment:
I don't fully understand the associated issue, but this is very misleading ... you are calling something `HW` which is actually the product of `W.dot(H)`.

Reply:
In that case, I should only replace WH as HW in function definition I guess. But this doesn't solve the issue's intentions, will that be fine?

Reply:
maybe @amueller has a suggestion off the top of his head? Otherwise I guess you will need to be familiar with NMF (maths, doc and code) to figure out what is the right thing to do.